osu.Game.Rulesets.Catch/Difficulty/CatchDifficultyCalculator.cs

```diff
@@ -34,7 +34,7 @@ public CatchDifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         {
         }
 
-        protected override DifficultyAttributes Calculate(IBeatmap beatmap, Mod[] mods, double timeRate)
+        protected override DifficultyAttributes Calculate(double upTo, IBeatmap beatmap, Mod[] mods, double timeRate)
```

Comment:
I'd make `upTo` the last parameter here and elsewhere, and give it a default parameter of `double.PositiveInfinity`.

Reply:
sounds like an improvement, for sure.